### ISSUE TITLE
Add support for custom CA for management clients

### ISSUE THREAD

**Issue by rajashekhargundeti** [open-policy-agent/opa#1954]
The SSL Cert for bundle provider is valid for api-uam-goclouddev.itcs.hpecorp.net.

I need to use localhost for testing.

I'm getting this error in the logs.
{"level":"error","msg":"Bundle download failed: request failed: Get https://api-uam-gocloud-llb.itcs.hpecorp.net/opa/data: x509: certificate is valid for api-uam-goclouddev.itcs.hpecorp.net, not localhost","name":"uam_policy","plugin":"bundle","time":"2019-12-12T08:44:26Z"}

Do we have any config parameter to skip SSL cert validation checks like "curl -k"

**Comment by patrick-east** [comment]
On the service configuration you should be able to specify `allow_insecure_tls`, that might be enough.

**Comment by rajashekhargundeti** [comment]
Thank you. This works. I couldn't find the same config parameter in documentation.

We have another scenario where we have internal CA who signs certs for bundle provider. I see the following error when OPA contacts bundle provider, in spite of passing --tls-ca-cert-file like below:

opa run -s -c ./dist/source/bundle.config -a 127.0.0.1:8004 --tls-ca-cert-file ./dist/ssl/cacert.pem

error is:
{"level":"error","msg":"Bundle download failed: request failed: Get https://api-uam-goclouddev.itcs.hpecorp.net/opa/data: x509: **certificate signed by unknown authority**","name":"uam_policy","plugin":"bundle","time":"2019-12-11T15:59:54Z"}

**Comment by patrick-east** [comment]
That `--tls-ca-cert-file` option is, i believe, for the OPA server as it serves requests to verify client certificates.

For the bundle downloads I don't think we have an option for it (at least not that I see). I think for now you would need to add the CA to the normal system trusted CA's for the host OPA is running on.

**Comment by tsandall** [comment]
As @patrick-east mentioned we do not currently support custom CAs for the mgmt clients (bundles, decision logs, etc): https://www.openpolicyagent.org/docs/latest/configuration/#client-tls-certificate

Extending the configuration and client to support can be added. @patrick-east could you provide a pointer about how someone could work on this and contribute it?

**Comment by patrick-east** [comment]
You basically just need to add a new field on the `rest.Config` struct for http clients https://github.com/open-policy-agent/opa/blob/master/plugins/rest/rest.go#L33-L43 Then make sure it is set correctly when initializing a client (probably over here https://github.com/open-policy-agent/opa/blob/master/plugins/rest/rest_auth.go#L24 in the `defaultTLSConfig` helper).

What we may want to do is configure it so that you can have either only the specified cert or include system CA's too (ie, when making the root CA pool whether we use `x509.SystemCertPool()` and add the specified one, or if we make an empty pool `x509.NewCertPool()` and add the specified one.

**Comment by ashish246** [comment]
Hi @tsandall @patrick-east , just following up on this one if any progress was made here. We, at ANZ, had a similar requirement where OPA container, deployed on-prem, is not able to talk to DAS, deployed on-cloud/GCP, because we are not able to install custom CA certificates in OPA container. Using `allow_insecure_tls` may not be a preferred way to move forward for us.

In our case though, OPA is pulling bundles from GCS buckets but failing to push status and decisions batch to DAS. We are receiving the same error as mentioned earlier in this ticket - 

``` 
"level":"error","msg":"Status update failed: Post \"https://styra-das.csp.gcpnp.anz/v1/status\": x509: certificate signed by unknown authority.","plugin":"status","time":"2021-06-04T00:55:58Z"}
```
Let me know if you need any further details. Cheers!

**Comment by tsandall** [comment]
@ashish246 this should be available now with #3575 merged. If you can give it a try before we cut the release, that would be excellent!
